app/actions/GalleryPictureActions.js

```diff
@@ -30,7 +33,7 @@ export function fetch(
         },
         propagateError: true,
       })
-    );
+    ).then(() => SelectGalleryPicturesByGalleryId(getState(), { galleryId }));
```

Comment:
I think this should be moved into the `downloadNext` function inside `Gallerydetail`, as it is only useful for that call, and is unnecessary for the other uses of fetch.

Reply:
Mathias Oterhals Myklebust
Good point! I was not able to move the selector to GalleryDetail, as this component does not have access to getState(). But I have now changed it to use the existing pictures prop instead, as this seems to be in sync. 
In addition, I have added a gallery clear action. This is because the image URLs expire one minute after retrieval. So if the user has had the page open for longer than this, the download will fail for the already fetched images. Let me know if this is a suitable solution.

Reply:
Good point! I was not able to move the selector to GalleryDetail, as this component does not have access to getState(). But I have now changed it to use the existing pictures prop instead, as this seems to be in sync. 
In addition, I have added a gallery clear action. This is because the image URLs expire one minute after retrieval. So if the user has had the page open for longer than this, the download will fail for the already fetched images. Let me know if this is a suitable solution.